gpt_engineer/core/ai.py

```diff
@@ -140,17 +102,12 @@ def __init__(self, model_name="gpt-4", temperature=0.1, azure_endpoint=""):
         """
         self.temperature = temperature
         self.azure_endpoint = azure_endpoint
-        self.model_name = model_name
+        self.model_name = self._check_model_access_and_fallback(model_name)
 
         self.llm = self._create_chat_model()
-        self.tokenizer = get_tokenizer(self.model_name)
-        logger.debug(f"Using model {self.model_name} with llm {self.llm}")
+        self.token_usage_log = TokenUsageLog(model_name)
 
-        # initialize token usage log
-        self.cumulative_prompt_tokens = 0
-        self.cumulative_completion_tokens = 0
-        self.cumulative_total_tokens = 0
-        self.token_usage_log = []
+        logger.debug(f"Using model {self.model_name} with llm {self.llm}")
```

Comment:
Aren't model and llm the same concepts?

Reply:
I have no memory of writing this... mustve been chat GPT 😄 . good catch

Reply:
Oh wait i didnt add this it was already there.. Ill clean it up to just model name